src/Imath/ImathMatrixAlgo.h

```diff
@@ -1125,7 +1125,8 @@ computeRSMatrix (
     const Matrix44<T>& B)
 {
     Vec3<T> as, ah, ar, at;
-    extractSHRT (A, as, ah, ar, at);
+    if (!extractSHRT (A, as, ah, ar, at))
+        throw std::domain_error ("degenerate A matrix in computeRSMatrix");
```

Comment:
I wonder about this pattern instead, in the interest of DRY? i.e. don't re-report the error reporting that has been generated by extractSHRT?

```suggestion
    try  { !extractSHRT (A, as, ah, ar, at)) } catch(...) { throw; }
```

Reply:
The compiler warning comes from the fact that extractSHRT() leaves the output parameters untouched if the input parameters are degenerate, and the compiler is smart enough to detect a code path in which the uninitialized values get used but not smart enough to recognize that an exception would get thrown to prevent that from happening.  So, it's not a real problem, just a confusing warning. To my reading, my proposed change leaves the code in the most understandable form.